### ISSUE TITLE
Replace Collect transform with mark sort

### ISSUE THREAD

**Issue by arvind** [vega/vega-lite#2669]
See vega/vega#930. While the Collect transform will sort input data tuples correctly, its sort order may not persist past the data join. Instead, we should be setting the `sort` property of the area/line marks. 

This is can be seen with the following Vega-Lite spec. If we add `"sort": {"field": "x"},` to the definition of the first area mark, the glitch is resolved.

![areaglitches](https://user-images.githubusercontent.com/42262/28292231-079bdb84-6b03-11e7-9118-e05f5a4d480e.gif)

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
  "data": {"url": "data/sp500.csv"},
  "vconcat": [{
    "width": 480,
    "transform": [{"filter": {"selection": "brush"}}],
    "mark": "area",
    "encoding": {
      "x": {
        "field": "date",
        "type": "temporal",
        "axis": {"title": "", "labelAngle": 0}
      },
      "y": {"field": "price","type": "quantitative"}
    }
  },
  {
    "width": 480,
    "height": 60,
    "mark": "area",
    "selection": {
      "brush": {"type": "interval", "encodings": ["x"]}
    },
    "encoding": {
      "x": {
        "field": "date",
        "type": "temporal",
        "axis": {"format": "%Y", "labelAngle": 0}
      },
      "y": {
        "field": "price",
        "type": "quantitative",
        "axis": {"tickCount": 3, "grid": false}
      }
    }
  }]
}
```

**Comment by kanitw** [comment]
Yeah, I'm glad now we have mark `sort` property in Vega.  
(This code path was written before mark's `sort` exists.)

**Comment by kanitw** [comment]
Will be fixed in #2750
